File: lobwebapp-core/src/main/java/com/espindola/lobwebapp/service/contract/base/EntityService.java
package com.espindola.lobwebapp.service.contract.base;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.security.access.prepost.PreAuthorize;

import com.espindola.lobwebapp.domain.base.AbstractEntity;
import com.espindola.lobwebapp.exception.InvalidArgumentException;
import com.espindola.lobwebapp.exception.NotFoundException;

//@PreAuthorize("hasRole('ROLE_USER')")
public interface EntityService<T extends AbstractEntity> {

	T save(T entity) throws NotFoundException, InvalidArgumentException;

	T update(T entity) throws NotFoundException, InvalidArgumentException;

	T remove(Long id) throws NotFoundException;

	T find(Long id) throws NotFoundException;

	List<T> findAll();

	Page<T> findAll(Pageable p);

	boolean exists(Long id);
}

File: lobwebapp-client/src/app/entity/service/EntityService.ts
///<reference path="../../reference.ts"/>

module entity {
    export interface EntityService<T extends AbstractEntity> {
        save(entity: T,
            successCallback: (data: T, status: number, headers: (headerName: string) => string) => any,
            errorCallback: (data: ValidationMessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;

        update(entity: T,
            successCallback: (data: T, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: ValidationMessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;

        remove(entity: T,
            successCallback: (data: T, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;

        find(id: number,
            successCallback: (data: T, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;

        list(successCallback: (data: T[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;

        list(successCallback: (data: T[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            pageable: Page): void;

        exists(entity: T,
            successCallback: (data: boolean, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;

        default(): T;
    }
}
